packages/general/test/environment/EnvironmentTest.ts

```diff
@@ -0,0 +1,999 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Environment } from "#environment/Environment.js";
+import { Environmental } from "#environment/Environmental.js";
+
+class TestService {
+    static [Environmental.create](environment: Environment) {
+        return environment.get(TestService);
+    }
+
+    closed = false;
+
+    close() {
+        this.closed = true;
+    }
+}
+
+class AnotherService {
+    static [Environmental.create](environment: Environment) {
+        return environment.get(AnotherService);
+    }
+
+    closed = false;
+
+    close() {
+        this.closed = true;
+    }
+}
+
+describe("Environment", () => {
+    describe("basic service management", () => {
+        it("creates and retrieves a service", () => {
+            const env = new Environment("test");
+            const service = new TestService();
+            env.set(TestService, service);
```

Comment:
These three lines repeat a lot.  Maybe put env and service in variables set up by a before each hook.  The few tests that need a different pattern can just ignore